libs/cardano-ledger-api/src/Cardano/Ledger/Api.hs

```diff
@@ -0,0 +1,100 @@
+-- | This module provides a library interface for working with types that will allow a user to
+-- interact with Cardano.
+--
+-- It is intended to be the complete API covering everything but without exposing constructors that
+-- reveal any lower level types.
+--
+-- In the interest of simplicity it glosses over some details of the system.
+-- Most simple tools should be able to work just using this interface, however you can go deeper and
+-- expose the types from the underlying libraries using "Cardano.Ledger.Core",
+-- "Cardano.Ledger.Shelley", "Cardano.Ledger.Babbage", etc.
+module Cardano.Ledger.Api
```

Comment:
It is a good idea to have the top level `Cardano.Ledger.Api` module that exports everything that is needed for working with ledger. However, it should re-export other module form the `cardano-ledger-api`, rather than export individual types with haddock. This is approximately how it suppose to look like:

```haskell
module Cardano.Ledger.Api (
  module Cardano.Ledger.Api.Era,
  module Cardano.Ledger.Api.Tx,
) where

import Cardano.Ledger.Api.Era
import Cardano.Ledger.Api.Tx
```

```haskell
module Cardano.Ledger.Api.Tx (
  module Cardano.Ledger.Api.Tx.Body,
  module Cardano.Ledger.Api.Tx.AuxData,
  module Cardano.Ledger.Api.Tx.Wits,
  ... all of Tx related stuff
) where

import Cardano.Ledger.Api.Tx.Body
-- these below aren't implemented yet, but should be here
import Cardano.Ledger.Api.Tx.AuxData
import Cardano.Ledger.Api.Tx.Wits
```

```haskell
module Cardano.Ledger.Api.Tx.Body (
  module Cardano.Ledger.Api.Tx.Out,
  module Cardano.Ledger.Api.Scripts,
  ... all of TxBody related stuff
) where

import Cardano.Ledger.Api.Tx.Out
import Cardano.Ledger.Api.Scripts
```

```haskell
module Cardano.Ledger.Api.Tx.Out (
  module Cardano.Ledger.Api.Tx.Out,
  module Cardano.Ledger.Api.Scripts.Data,
  ... all of TxOut related stuff
) where

import Cardano.Ledger.Api.Tx.Out
import Cardano.Ledger.Api.Scripts.Data
```

If it is not done this way, then we will either end up duplicating documentation in all places or one of modules will miss the relevant documentation. But if we do module re-exporting then all of haddocks and examples stay inside their respectful modules. Another problem that it solves: it is way too easy to export a new function or a type from one module, but not the top level. So, this hierarchy is pretty important.


I can finish up the full module hierarchy, but since you added the `Cardano.Ledger.Api`, could you re-export existing modules in the suggested way and move the example into the `Cardano.Ledger.Api.Tx` module.